### PR TITLE
[enhancement](auto-partition) Forbid use Auto and Dynamic partition at the same time

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

1. Forbid use Auto and Dynamic partition at the same time
2. Fix lost check in legacy planner for creating auto partition table

doc pr: https://github.com/apache/doris-website/pull/563

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java [hunk 1]
```diff
@@ -2729,21 +2729,8 @@ private void createOlapTable(Database db, CreateTableStmt stmt) throws UserExcep
...
-                                    String autoUnit = ((LiteralExpr) child).getStringValue();
-                                    if (!dynamicUnit.equalsIgnoreCase(autoUnit)) {
-                                        throw new AnalysisException(
-                                                "If support auto partition and dynamic partition at same time, "
-                                                        + "they must have the same interval unit.");
-                                    }
-                                }
-                            }
-                        }
+                        throw new AnalysisException(
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:2732]: why the Execption in two different part ?

Comment (zclllyybb) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/InternalCatalog.java:2732]: It's legacy and nereids planner

**Comment by zclllyybb** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**